File: MyPortal.Core/Entities/TrainingCourse.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MyPortal.Core.Entities
{
    [Table("TrainingCourses")]
    public class TrainingCourse : LookupEntity
    {
        [Required]
        [StringLength(128)]
        public string Code { get; set; } = null!;

        [Required]
        [StringLength(128)]
        public string Name { get; set; } = null!;
    }
}

File: MyPortal.Frontend/myportal-spa/src/app/shared/types/training-course.ts
export interface TrainingCourse {
  id: string;
  code: string;
  name: string;
  description: string | null;
  active: boolean;
  inUse: boolean;
}

export interface TrainingCourseUpsert {
  code: string;
  name: string;
  description: string | null;
  active: boolean;
}
